sdk/keyvault/azure-keyvault-secrets/samples/hello_world.py

```diff
@@ -43,28 +45,30 @@
 print("\n.. Create Secret")
 expires = datetime.datetime.utcnow() + datetime.timedelta(days=365)
 secret = client.set_secret("helloWorldSecretName", "helloWorldSecretValue", expires_on=expires)
+assert secret.name
 print(f"Secret with name '{secret.name}' created with value '{secret.value}'")
 print(f"Secret with name '{secret.name}' expires on '{secret.properties.expires_on}'")
 
 # Let's get the bank secret using its name
 print("\n.. Get a Secret by name")
 bank_secret = client.get_secret(secret.name)
+assert bank_secret.properties.expires_on
 print(f"Secret with name '{bank_secret.name}' was found with value '{bank_secret.value}'.")
 
 # After one year, the bank account is still active, we need to update the expiry time of the secret.
 # The update method can be used to update the expiry attribute of the secret. It cannot be used to update
 # the value of the secret.
 print("\n.. Update a Secret by name")
-expires = bank_secret.properties.expires_on + datetime.timedelta(days=365)
+expires = cast(datetime.datetime, bank_secret.properties.expires_on + datetime.timedelta(days=365))
```

Comment:
If we update the `expires_on` typing will this work without the cast? or is this a side effect of being optional?

Reply:
This was actually a surprising warning from `mypy`, saying that the assignment was invalid because the right side was a `Union[datetime, timedelta]`:
```
error: Incompatible types in assignment (expression has type "Union[datetime, timedelta]", variable has type "datetime")  [assignment]
```